_core/smart/tutorial.md

```diff
@@ -0,0 +1,335 @@
+
+Blockstack publishes an Clarity's developer preview image on Docker hub. This virtual environment (VM) that contains sample programs, the Blockstack Core, and tools for workign with them. In this task, you use Docker to pull and and run the image on your local workstation. 
+
+1. Pull the Blockstack core `clarity-developer-preview` image from Docker Hub.
+
+    ```bash
+    $ docker pull blockstack/blockstack-core:clarity-developer-preview
+    ``` 
+
+2. Start the Blockstack Core test environment with a Bash shell.
+
+    ```bash
+    $ docker run -it -v $HOME/blockstack-dev-data:/data/ blockstack/blockstack-core:clarity-developer-preview bash
+    ```
+
+    The launches a container with the Clarity VM and opens a bash shell into the VM. The `-v` flag creates a local `$HOME/blockstack-dev-data` directory in your workstation and mounts it at the `/data` directory inside the container. The shell opens into the `src/blockstack-core` directory. This directory contains the source for a core and includes Clarity contract samples you can run.
+
+3. List the contents of the `sample-programs` directory.
+
+   ```bash
+   root@f88368ba07b2:/src/blockstack-core# ls sample-programs/
+   names.clar  tokens.clar
+   ```
+
+   The sample programs directory contains two simple Clarity programs. Clarity code files have a `.clar` suffix.
+
+4. Go ahead and display the contents of the `tokens.clar` program with the `cat` command.
+
+    ```bash
+    root@c28600552694:/src/blockstack-core# cat sample-programs/tokens.clar 
+    ```
+
+    The next section gives you an introduction to the Clarity language by way of examining this program's code.
+
+## Task 2: Review a simple Clarity program
+
+If you haven't already done so, use the `cat` or `more` command to dipslay the `tokens.clar` file's code. Clarity is designed for static analysis; it is not a compiled language and is not Turing complete. It language is a LISP-like language.  LISP is an acronym for list processing. the first line of the program contains a user-defined `get-balance` function.  
+
+```cl
+(define (get-balance (account principal))
+  (default-to 0 (get balance (fetch-entry tokens (tuple (account account))))))
+```
+
+`get-balance` is a private function because it is constructed with the `define` call. To create public functions, you would use the `define-public` function. Public functions can be called from other contracts or even from the command line with the `clarity-cli`.
+
+Notice the program is enclosed in  `()` (parentheses) and each statement as well.  The `get-balance` function defines an `account` field which takes the special type `principal`, Principals represent a spending entity and are roughly equivalent to a Stacks address. 
+
+Along with the `principal` types, Clarity supports  booleans, integers, and fixed length buffers. Variables are created via `let` binding but there is no support for mutating functions like `set`.
+The `names.clar` code includes two public functions.  
```

Comment:
> The `names.clar` code includes two public functions.  
This seems out of place --I think we're still looking at `tokens.clar` at this point.

Reply:
✔️